docs/js/sortable-table.js

```diff
@@ -0,0 +1,74 @@
+import { onDomReady } from "@minvws/manon/utils.js";
+
+onDomReady(initSortableTable);
+
+function initSortableTable() {
+  var thead = document.querySelector("#sortable-table-example thead");
+  var tbody = document.querySelector("#sortable-table-example tbody");
+  if (!(thead instanceof HTMLTableSectionElement) || !(tbody instanceof HTMLTableSectionElement)) {
+    return;
+  }
+  thead.addEventListener("click", function(event) {
+    var button = event.target;
+    if (!(button instanceof HTMLButtonElement)) {
+      return;
+    }
+    var sortStatus = getSortStatus(thead);
+    var columnIndex = getColumnIndex(button.parentElement);
+    if (columnIndex === sortStatus.columnIndex) {
+      sortStatus.direction = sortStatus.direction === 'ascending' ? 'descending' : 'ascending';
+    } else {
+      sortStatus.columnIndex = columnIndex;
+      sortStatus.direction = 'ascending';
+    }
+    updateSortButtons(thead, sortStatus);
+    sortTableBody(tbody, sortStatus);
+  });
+}
+
+function getColumnIndex(th) {
+  return Array.prototype.indexOf.call(th.parentElement.children, th);
+}
+
+function getSortStatus(thead) {
+  var sorted = thead.querySelector("[aria-sort]");
+  if (!sorted) {
+    return { columnIndex: -1, direction: null };
+  }
+  return {
+    columnIndex: getColumnIndex(sorted),
+    direction: sorted.getAttribute("aria-sort")
+  }
+}
+
+function updateSortButtons(thead, sortStatus) {
+  var ths = thead.querySelectorAll("th");
+  var sorted = thead.querySelector("th[aria-sort]");
+  if (sorted && getColumnIndex(sorted) !== sortStatus.columnIndex) {
+    sorted.removeAttribute("aria-sort");
+    var icon = sorted.querySelector(".icon");
+    if (icon) {
+      icon.classList.remove("icon-sortable", "icon-ascending", "icon-descending");
+      icon.classList.add("icon-sortable");
+    }
+  }
+  var sort = ths[sortStatus.columnIndex];
+  if (sort) {
+    sort.setAttribute("aria-sort", sortStatus.direction);
+    var icon = sort.querySelector(".icon");
+    if (icon) {
+      icon.classList.remove("icon-sortable", "icon-ascending", "icon-descending");
+      icon.classList.add("icon-" + sortStatus.direction);
+    }
+  }
+}
+
+function sortTableBody(tbody, sortStatus) {
+  var cells = tbody.querySelectorAll("tr > td:nth-of-type(" + (sortStatus.columnIndex + 1) + ")");
+  var comparator = sortStatus.direction === "ascending"
```

Comment:
Should we also set the numeric=true option for columns that are rendered as such?

Reply:
Yes. At the moment the docs explicitly leave the JS implementation to the reader. This simple interactive example to accompany the docs doesn't have a numeric column, but it makes sense to add one, just in case someone goes snooping around in the JS. And if we do include a more in-depth implementation guide as part of the docs it makes sense to address the nuances of sorting there, too.

Reply:
I've added `class="number"` to the example (source & snippet) and used a `data-`attribute on the `<th>` for the example implementation, but I've left that out of the the snippet so as not to give the impression that that's "the" way to do it.